packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -0,0 +1,112 @@
+   */
+  readonly project: codebuild.IProject;
+  /**
+   * A set of environment variables that overrides, for this build only,
+   * the latest ones already defined in the build project.
+   *
+   * @default No override
+   */
+  readonly environmentVariablesOverride?: { [name: string]: codebuild.BuildEnvironmentVariable };
+}
+
+/**
+ * Start a CodeBuild Build as a task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-codebuild.html
+ */
+export class CodeBuildStartBuild extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: CodeBuildStartBuildProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, CodeBuildStartBuild.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskMetrics = {
+      metricPrefixSingular: 'CodeBuildProject',
+      metricPrefixPlural: 'CodeBuildProjects',
+      metricDimensions: {
+        ProjectArn: this.props.project.projectArn,
+      },
+    };
+
+    this.taskPolicies = this.configurePolicyStatements();
+  }
+
+  private configurePolicyStatements(): iam.PolicyStatement[] {
+    let policyStatements = [
+      new iam.PolicyStatement({
+        resources: [this.props.project.projectArn],
+        actions: [
+          'codebuild:StartBuild',
+          'codebuild:StopBuild',
+        ],
+      }),
+    ];
+
+    if (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      policyStatements.push(
+        new iam.PolicyStatement({
+          actions: ['events:PutTargets', 'events:PutRule', 'events:DescribeRule'],
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: 'events',
+              resource: 'rule/StepFunctionsGetEventForCodeBuildStartBuildRule',
+            }),
+          ],
+        }),
+      );
+    }
+
+    return policyStatements;
+  }
+
+  /**
+   * Provides the CodeBuild StartBuild service integration task configuration
+   */
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('codebuild', 'startBuild', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        ProjectName: this.props.project.projectName,
+        EnvironmentVariablesOverride: this.props.environmentVariablesOverride
+          ? codebuild.Project
+            .serializeEnvVariables(this.props.environmentVariablesOverride)
+            .map((environmentVariable: codebuild.CfnProject.EnvironmentVariableProperty) => {
```

Comment:
I see what you mean. I was just trying to reuse the existing `serializeEnvVariables` from the `codebuild.Project` class. It's a small method and I can probably go with writing my own and it's going to avoid the round trip.

See the implementation below.

```
private serializeEnvVariables(environmentVariables: { [name: string]: codebuild.BuildEnvironmentVariable }) {
    return Object.keys(environmentVariables).map(name => ({
      Name: name,
      Type: environmentVariables[name].type || codebuild.BuildEnvironmentVariableType.PLAINTEXT,
      Value: environmentVariables[name].value,
    }));
  }
```